com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -702,16 +707,25 @@ void AgentStep()
             if ((m_RequestAction) && (m_Brain != null))
             {
                 m_RequestAction = false;
-                if (m_Action.vectorActions != null)
-                {
-                    AgentAction(m_Action.vectorActions);
-                }
+                AgentAction(m_Action.vectorActions);
             }
         }
 
         void DecideAction()
         {
             m_Action.vectorActions = m_Brain?.DecideAction();
```

Comment:
I'm guessing this is set to `null` on reset?

Reply:
Yeah, I think that might be better; ResetData() resets the stored actions, but this change keeps them (possibly with stale data).

Reply:
We are not really reseting the data, we are generating a vector of zeros in the case the action is null

Reply:
Alright, changed it since there seems to be consensus.

Reply:
Then how about moving the vectorAction code from ResetData to a new method (ResetActions()?), call that from ResetData, and call it here?

Reply:
Sorry, missed your most recent comment before I made mine.